src/Streamly/Internal/LZ4.hs

```diff
@@ -257,11 +257,11 @@ decompressChunk ::
 decompressChunk ctx arr = do
     Array.asPtr arr
         $ \src -> do
-              let hdrDecompLen :: Ptr Int32 = castPtr src
-                  hdrSrcLen :: Ptr Int32 = src `plusPtr` 4
-                  compData = src `plusPtr` 8
-                  arrDataLen = Array.byteLength arr - 8
-              decompLen <- coerce <$> peek hdrDecompLen
+              let -- hdrDecompLen :: Ptr Int32 = castPtr src
+                  hdrSrcLen :: Ptr Int32 = src
+                  compData = src `plusPtr` 4
+                  arrDataLen = Array.byteLength arr - 4
+              decompLen <- return $ 1024 * 1024 * 2
```

Comment:
This is the source of the error.
If the `decompLen` is not exactly `decompLen` then there is memory corruption.
`shrinkToFit` is not shrinking the array in place probably.